doc/mgmt/SONiC_AliasFeature_HLD.md

```diff
@@ -0,0 +1,337 @@
+| For any CRU request that reaches Xfmr, Xfmr-infra will read data from db and  create dbDataMap.  It then invokes overloaded  functions(fieldxfmr, subtreexfmr…) with dbDataMap. Fills the ygot tree with  data from dbDataMap & merges it with ygot-tree from subtree-xfmr and sends response back. | For any CRU request that reaches Xfmr, xfmr-infra will read data from db and create dbDataMap. First value-xfmr (if present) is invoked, dbDataMap is processed and a new processed-dbDataMap  is created.  Overloaded  functions(fieldxfmr, subtreexfmr…) with processed-dbDataMap will be invoked next. Fills the ygot tree with  data from dbDataMap & merges it with ygot-tree from subtree-xfmr and sends response back. |
+
+   **Sub-tree Transformer**: In cases where applications use subtree transformer, then need to use an API that will return the alias string. The API will internally figure out if alias is enabled or not.
+
+   Following are the APIs used:  
+   These APIs are available in `translib/utils` package.
+
+   **GetNativeNameFromUIName**  
+   `func GetNativeNameFromUIName(ifName *string) *string`  
+
+   Retrieves native interface name from user input name, if the alias mode is enabled. User input name can be alias or native interface. API will provide native interface name if the user input is alias name, otherwise it will return the native interface name passed to the API. If the alias mode is not enabled, API will return the input string passed.
+
+   **GetAliasNameFromUIName**  
+   `func GetAliasNameFromUIName(ifName *string) *string`  
+
+   Retrieves alias interface name from user input name, if the alias mode is enabled. User input name can be alias or native interface. API will provide alias name if the user input is native interface name, otherwise it will return the alias name passed to the API. If the alias mode is not enabled, API will return the input string passed.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+
+#### 3.2.1.1
+
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+Yang model used for alias mode handling  
+[sonic-device-metadata.yang](https://github.com/project-arlo/sonic-mgmt-framework/blob/master/models/yang/sonic/sonic-device-metadata.yang)  
+
+Supported yang objects and attributes are highlighted in green:
+```diff
+module: sonic-device-metadata
+    +--rw sonic-device-metadata
+       +--rw DEVICE_METADATA
+          +--rw DEVICE_METADATA_LIST* [name]
+             +--rw name                          string
++            +--rw aliasMode                     boolean
+```
+
+### 3.6.2 CLI
+
+#### 3.6.2.1 Configuration Commands
+
+#### Alias mode configuration
+
+`[no] alias enable`
+```
+sonic(config)# alias enable

```

Comment:
Generally “alias” is a command to create aliases for existing commands. So for extensibility and to avoid confusion with other vendors command, we should consider changing it to something like this.
o	[no] alias eth-intf-name enable